[iOS Debug] 2 MediaRecorder tests are constantly crashing due to assertion failure.

jyavenard opened the pull request:
#### 0a1ac6a9346a53e4ec3d65532e9ea6950f7a99a2
<pre>
[iOS Debug] 2 MediaRecorder tests are constantly crashing due to assertion failure.
<a href="https://bugs.webkit.org/show_bug.cgi?id=290410">https://bugs.webkit.org/show_bug.cgi?id=290410</a>
<a href="https://rdar.apple.com/147863063">rdar://147863063</a>

Reviewed by Youenn Fablet.

When sending the PlatformTimeRanges to the UI process, we converted each
TimeRange to a Vector&lt;pair&lt;double, double&gt;&gt;. When the TimeRanges contained
+infinity, the conversion from MediaTime::positiveInfinity -&gt; double -&gt; MediaTime were lossy
and caused the generated PlatformTimeRanges to fail the assertion `end.isValid()`

We remove this unnecessary TimesRanges to Vector when sending it over IPC
and instead use the standard serialiser.
In addition, stop using WebCore::TimeRanges and instead have
seekableRangesChanged() and seekableRanges() return PlatformTimeRanges object.
This also removes a slight layering violation that had existed for a long
time.

Covered by existing tests.

* LayoutTests/platform/ios/TestExpectations: Change test TestExpectations as tests won&apos;t cause an assertion on debug build anymore.
* Source/WebCore/platform/graphics/PlatformTimeRanges.cpp:
(WebCore::PlatformTimeRanges::add): Assert that the end is valid on all platform.

Canonical link: <a href="https://commits.webkit.org/294891@main">https://commits.webkit.org/294891@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e2e685e17ff19e76bc4eb8521b3810e478151264

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/103464 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/23146 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/13465 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/108638 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/54109 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/105503 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/23496 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/31697 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/78621 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/106470 "Build is in progress. Recent messages:OS: Sequoia (15.3), Xcode: 16.2; Running apply-patch; Checked out pull request; Passed webkitperl tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/18214 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/93331 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/58956 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/18034 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/11378 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/53465 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/87828 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/11438 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/111017 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/30611 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/22574 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/87616 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/30972 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/89531 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/87257 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22208 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/32135 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/9844 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/24953 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/30539 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/35851 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/30339 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/33670 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/31900 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

youennf commented on Source/WebCore/platform/graphics/MediaPlayerPrivate.cpp:
```diff
@@ -51,6 +51,7 @@ const PlatformTimeRanges& MediaPlayerPrivateInterface::seekable() const
 {
     if (maxTimeSeekable() == MediaTime::zeroTime())
```
I would tend to do: `auto maxTimeSeekable = this->maxTimeSeekable()` given this is a virtual method.

youennf commented on Source/WebCore/platform/graphics/MediaPlayerPrivate.cpp:
```diff
@@ -51,6 +51,7 @@ const PlatformTimeRanges& MediaPlayerPrivateInterface::seekable() const
         return PlatformTimeRanges::emptyRanges();
+    ASSERT(maxTimeSeekable().isValid());
     m_seekable = { minTimeSeekable(), maxTimeSeekable() };
     return m_seekable;
```
Why do we need `m_seekable` if we are recreating it each time?

jyavenard replied:
Because we return a reference here.

youennf reviewed (approved):
Great simplification.

jyavenard committed:
[iOS Debug] 2 MediaRecorder tests are constantly crashing due to assertion failure.
https://bugs.webkit.org/show_bug.cgi?id=290410
rdar://147863063

Reviewed by Youenn Fablet.

When sending the PlatformTimeRanges to the UI process, we converted each
TimeRange to a Vector<pair<double, double>>. When the TimeRanges contained
+infinity, the conversion from MediaTime::positiveInfinity -> double -> MediaTime were lossy
and caused the generated PlatformTimeRanges to fail the assertion `end.isValid()`

We remove this unnecessary TimesRanges to Vector when sending it over IPC
and instead use the standard serialiser.
In addition, stop using WebCore::TimeRanges and instead have
seekableRangesChanged() and seekableRanges() return PlatformTimeRanges object.
This also removes a slight layering violation that had existed for a long
time.

Covered by existing tests.

* LayoutTests/platform/ios/TestExpectations: Change test TestExpectations as tests won't cause an assertion on debug build anymore.
* Source/WebCore/platform/graphics/PlatformTimeRanges.cpp:
(WebCore::PlatformTimeRanges::add): Assert that the end is valid on all platform.

Canonical link: https://commits.webkit.org/294891@main
